config/v1/types_infrastructure.go

```diff
@@ -189,6 +192,45 @@ const (
 	IBMCloudProviderTypeUPI IBMCloudProviderType = "UPI"
 )
 
+type CloudControllerManagerState string
+
+const (
+	// Cloud Controller Manager is enabled and expected to be supplied.
+	// Signaling that kubelets and other CCM consumers should use --cloud-provider=external.
+	CloudControllerManagerExternal CloudControllerManagerState = "External"
+
+	// Cloud Controller Manager is enabled and expected to be supplied.
+	// Signaling that kubelets and other CCM consumers should not set --cloud-provider flag.
+	CloudControllerManagerNone CloudControllerManagerState = "None"
+)
+
+type CloudControllerManagerSettings struct {
```

Comment:
Nit, I think we would normally call this config rather than settings

New structs should have godoc explaining their usage/intention

Reply:
Idiomatically I think it should be `CloudControllerManagerSpec`, so lets make it that

Reply:
done